awscli/customizations/wizard/app.py

```diff
@@ -41,7 +41,8 @@ def run(self, loaded):
 
 class WizardApp(Application):
     def __init__(self, layout, values, traverser, executor, style=None,
-                 key_bindings=None, full_screen=True, output=None):
+                 key_bindings=None, full_screen=True, output=None,
+                 app_input=None):
```

Comment:
What were these `app_input` additions for?

Reply:
I had the rationale in the second commit message:
```
Add the option to prompt toolkit abstractions to inject an input. This was needed because the stdin
provided by pytest does not  have a fileno and as a result causes prompt toolkit to fail when instantiating
its default wrapper around stdin.
```
Basically we cannot instantiate a normal prompt toolkit `Application` class (which is the main class for prompt toolkit usage) when running tests so we need to inject a fake input in order to be able instantiate it.